src/dialects/abstract/query-generator.js

```diff
@@ -103,12 +113,14 @@ class QueryGenerator {
     _.defaults(options, this.options);
 
     const modelAttributeMap = {};
-    const bind = [];
+    const bind = options.bind || [];
```

Comment:
Yes, it is necessary, we are passing bind.length in line 265 to populateInsertQueryReturnIntoBinds function. If options.bind is undefined it would generate an error.

Reply:
options.bind is populated in upsertQuery from the result of updateQuery, which is used here to append additional binds.